datafusion/core/tests/parquet/page_pruning.rs

```diff
@@ -204,3 +222,466 @@ async fn page_index_filter_multi_col() {
     let batch = results.next().await.unwrap().unwrap();
     assert_eq!(batch.num_rows(), 7300);
 }
+
+async fn test_prune(
```

Comment:
This is great coverage -- thanks @Ted-Jiang. It is somewhat repetitive with the row group pruning but I think that is ok as they are different code paths